### DISCUSSION TITLE
How to apply the MERL-MIT-BRDF dataset in mitsuba

### DISCUSSION THREAD

**Discussion by ouyxi** [mitsuba-renderer/mitsuba3#1436] [Q&A] [answered]
I tried to apply the MERL-BRDF  dataset on mitsuba. I converted the binary file into a [90,90,180,3] matrix according to the dataset Readme file, but I don't know how to apply it on mitsuba. As a novice, this problem has troubled me for a long time.

**Comment by merlinND** [comment]
Since there is no BSDF plugin intended to directly read the values from the MERL database (that I'm aware of), perhaps the easiest way would be to convert this data into the format expected by the `measured` BSDF?

https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_bsdfs.html#measured-material-measured
https://github.com/mitsuba-renderer/mitsuba3/blob/bc0d50171861b04fe06454ad92c07ba7449c2069/src/bsdfs/measured.cpp

**Reply by ouyxi** [reply]
Thank you very much for your answer! I have tried this method and downloaded the dataset from https://rgl.epfl.ch/materials. I checked the dataset fields including ['version', 'description', 'phi_i', 'theta_i', 'sigma', 'ndf', 'vndf', 'luminance', 'rgb', 'jacobian', 'valid']. 
However, it is still difficult to convert the MERL-BRDF dataset to the same format. The .bsdf file I converted only contains the 'rgb' field. It is speculated that because the MERL dataset only contains floating point numbers, it is uniformly converted to the 'rgb' field, but I don't know how to avoid this. Maybe I need to compile a plugin specifically for the MERL dataset later.

**Reply by merlinND** [reply, marked as answer]
If the formats don't easily match, then indeed it might be best to use a dedicated custom plugin.

You could have a look at the plugin posted by @fzy28:
https://github.com/mitsuba-renderer/mitsuba3/issues/1011#issue-2042062890 (see also the rest of the discussion in that thread)

Please make sure to check the license of the code with them before using it.

**Comment by ouyxi** [comment]
Thank you for your reply！His solution gave me new ideas!
